mlflow/tracking/_default_experiment/databricks_job_experiment_provider.py

```diff
@@ -0,0 +1,36 @@
+import logging
+from mlflow.tracking._default_experiment.abstract_context import DefaultExperimentProvider
+from mlflow.utils import databricks_utils
+from mlflow.tracking.client import MlflowClient
+from mlflow.utils.mlflow_tags import (
+    MLFLOW_DATABRICKS_JOB_TYPE_INFO,
+    MLFLOW_EXPERIMENT_SOURCE_TYPE,
+    MLFLOW_EXPERIMENT_SOURCE_ID,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class DatabricksJobExperimentProvider(DefaultExperimentProvider):
+    def in_context(self):
+        return (
+            databricks_utils.is_in_databricks_job()
+            and databricks_utils.get_job_type_info() == "NORMAL"
+        )
+
+    def get_experiment_id(self):
+        job_id = databricks_utils.get_job_id()
+        tags = {}
+        tags[MLFLOW_DATABRICKS_JOB_TYPE_INFO] = databricks_utils.get_job_type_info()
+        tags[MLFLOW_EXPERIMENT_SOURCE_TYPE] = "JOB"
+        tags[MLFLOW_EXPERIMENT_SOURCE_ID] = job_id
+
+        experiment_id = MlflowClient().create_experiment(
```

Comment:
Can we add an inline comment that this is a "get or create" operation on Databricks when the `JOB` source type is specified? This will help future readers of the code understand what's happening.

Reply:
No strong preference here, but isn't "get or create" not the client's concern? The client is just asking the backend to create an experiment and give it an ID to work with; details of where exactly that ID came from are abstracted away.

Reply:
It's our concern as maintainer's of the Databricks Jobs + Experiments integration, since we've gone ahead and implemented undocumented backend API behavior that a reader of this client code wouldn't otherwise be able to understand without sifting through the implementation of the Databricks MLflow Tracking service. All I'm asking for is some clarifying context.